web/modules/custom/joinup_news/src/Guard/FulfillmentGuard.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+namespace Drupal\joinup_news\Guard;
+
+use Drupal\og\Og;
+use Drupal\state_machine\Guard\GuardInterface;
+use Drupal\state_machine\Plugin\Workflow\WorkflowInterface;
+use Drupal\state_machine\Plugin\Workflow\WorkflowTransition;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Guard class for the transitions of the news entity.
+ *
+ * @package Drupal\joinup_news\Guard
+ */
+class FulfillmentGuard implements GuardInterface {
+  /**
+   * Constant representing the pre-moderating parent.
+   */
+  const MODERATED = 1;
+
+  /**
+   * Constant representing the post-moderating parent.
+   */
+  const NON_MODERATED = 0;
+
+  /**
+   * {@inheritdoc}
+   *
+   * We need to override default transitions allowed because this is also
+   * dependant on the parent's moderation, system roles and organic groups
+   * user roles. In the following method, the allowed transitions per
+   * moderation are checked and then if the transition is allowed, the user
+   * roles by the system and the organic groups are checked.
+   */
+  public function allowed(WorkflowTransition $transition, WorkflowInterface $workflow, EntityInterface $entity) {
```

Comment:
Check here if the guard applies:
Only 'nodes' of the type 'news' should be handled.